qiskit/test/base.py

```diff
@@ -88,9 +89,106 @@ def gather_details(source_dict, target_dict):
         target_dict[name] = _copy_content(content_object)
 
 
-class BaseQiskitTestCase(unittest.TestCase):
```

Comment:
Oh yeah, of course.  I must have spaced out, since as it stands I'm removing public API components without so much as a deprecation notice.

I guess we should decide what should actually be the public API of our test cases.  At the moment, `QiskitTestCase` includes a few things that are specific to Terra - perhaps it would be better if this base class is available for _all_ Qiskit packages, and then Terra derives its own class `TerraTestCase` which adds on the package-level particulars of the logging, the warning filters, etc.  It adds an extra layer in the inheritance, but it'd probably be easier to read because there'd be a proper separation of concerns, and we'd be less likely to break consumers of the API when we want to modify Terra's tests.